### PR TITLE
Feature/homework 4

### PR DESCRIPTION
**Pull request by SergeyShatter**
https://docs.google.com/document/d/14UaH4FuqSkenaY8_mOdLGaAhwoTcaSwzvdDODymr4Lw/edit

### CODE REVIEW

File: routes/productsRoutes.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const productId = req.params.productId;
+
+  const product = products.find(item => item.id === productId);
+
+  res.status(200);
+
+  if (product) {
+    res.send(product);
+  } else {
+    res.send('Product not found!');
```
Comment (MaksimKavalenkaBY) [inline routes/productsRoutes.js]: If there is no product found, you can set 404

File: routes/productsRoutes.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+router.get('/products', (req, res) => {
+  res.status(200).send(products);
+});
+
+
+router.get('/products/:productId', (req, res) => {
+  const productId = req.params.productId;
+
+  const product = products.find(item => item.id === productId);
```
Comment (MaksimKavalenkaBY) [inline routes/productsRoutes.js]: It is better to move such kind of actions to a controller. For example, you'll have productController.js that provides methods such as getProducts, getProductById and etc. So the only controller will be worrying about a destination where this info is stored. And routes will be using the only methods of a specific controller.

File: routes/productsRoutes.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  if (product) {
+    res.send(product.reviews);
+  } else {
+    res.send('Product not found!');
+  }
+});
+
+
+router.post('/products', (req, res) => {
+  const { title } = req.parsedQuery;
```
Comment (MaksimKavalenkaBY) [inline routes/productsRoutes.js]: Well, it forces a user to split up a body and move some info into a query (like a _title_). I would insist on being more consistent and allowing the user to create a product through the body.

File: routes/productsRoutes.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+  if (!title) {
+    res.status(400).send('Product title is required!');
+    return;
+  } else {
+    const newProduct = {
+      title,
+      reviews: [],
```
Comment (MaksimKavalenkaBY) [inline routes/productsRoutes.js]: We should allow a user to use a body to initialize any attribute a product has (_reviews_ as well). In this case we should parse the body in order to extract that info. For this case just plug **json** and **urlencoded** functions as a middleware for _express_.
https://expressjs.com/en/guide/using-middleware.html#middleware.built-in

File: routes/productsRoutes.js (hunk 2 above)
Comment (SergeyShatter) [reply routes/productsRoutes.js]: Thanks for the detailed explanation!)

File: routes/productsRoutes.js [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import express from 'express';
+import { ProductsController } from '../controllers';
+
+const router = express.Router();
+
+
+router.get('/products', (req, res) => {
+  res.status(200).send(products);
```
Comment (MaksimKavalenkaBY) [inline routes/productsRoutes.js]: Please don't forget to add _ProductsController.getProducts()_ before.

**Pull request merged**